### PR TITLE
Fix the bumblebee document overlay date getter to allow for `None`

### LINKED ISSUE #4723: Bumblebee Overlay: Document without document_date causes MissingParameterError
https://sentry.4teamwork.ch/sentry/onegov-gever/issues/6425/

#### Steps to reproduce
- Add a document
- Edit the document and remove the `document_date`, therefore setting it to `None` (*don't be fooled by the fact that a date still gets displayed, that's a different bug related to default values)*
- Open the Bumblebee-Overlay by clicking on the preview thumbnail

```
MissingParameterError: Missing required parameter(s): datetime

 - Expression: "view/overlay/get_document_date"
 - Filename:   ... pengever/bumblebee/browser/templates/bumblebeeoverlay.pt
 - Location:   (line 10: col 27)
 - Source:     ... l:define="date view/overlay/get_document_date" tal:condition ...
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ViewPageTemplateFile - at 0x7f40ef440110>
               views: <ViewMapper - at 0x7f40dbed31d0>
               modules: <instance - at 0x7f410b057c20>
               args: <tuple - at 0x7f411be1e050>
               here: <ImplicitAcquisitionWrapper document-55286 at 0x7f40da8a2050>
               user: <ImplicitAcquisitionWrapper - at 0x7f40e101dc30>
               file_size: 529
               nothing: <NoneType - at 0x7ad050>
               container: <ImplicitAcquisitionWrapper document-55286 at 0x7f40da8a2050>
               request: <instance - at 0x7f40d8678518>
               wrapped_repeat: <SafeMapping - at 0x7f40d8b421b0>
               traverse_subpath: <list - at 0x7f40d9484c68>
               default: <object - at 0x7f411bdf4c20>
               loop: {...} (0)
               context: <ImplicitAcquisitionWrapper document-55286 at 0x7f40da8a2050>
               view: <SimpleViewClass from /home/zope/eggs/opengever.core-2017.6.8-py2.7.egg/opengever/bumblebee/browser/templates/bumblebeeoverlay.pt bumblebee-overlay-document at 0x7f40dbed3f10>
               translate: <function translate at 0x7f40e8ee4500>
               root: <ImplicitAcquisitionWrapper Zope at 0x7f40dab973c0>
               options: {...} (0)
               target_language: de
(13 additional frame(s) were not displayed)
...
  File "chameleon-cache/63b0c8a1adeff85525456b5bd81cfa46.py", line 232, in render
    __value = _static_139634488256592(getitem('view'), econtext, True, ('overlay', 'get_document_date', ))
  File "home/zope/eggs/five.pt-2.2.4-py2.7.egg/five/pt/expressions.py", line 161, in __call__
    return base()
  File "home/zope/eggs/opengever.core-2017.6.8-py2.7.egg/opengever/bumblebee/browser/overlay.py", line 68, in get_document_date
    return api.portal.get_localized_time(self.context.document_date)
  File "<string>", line 2, in get_localized_time
  File "home/zope/eggs/plone.api-1.7-py2.7.egg/plone/api/validation.py", line 67, in wrapped
    ', '.join(missing))

MissingParameterError: Missing required parameter(s): datetime

 - Expression: "view/overlay/get_document_date"
 - Filename:   ... pengever/bumblebee/browser/templates/bumblebeeoverlay.pt
 - Location:   (line 10: col 27)
 - Source:     ... l:define="date view/overlay/get_document_date" tal:condition ...
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ViewPageTemplateFile - at 0x7f40ef440110>
               views: <ViewMapper - at 0x7f40dbed31d0>
               modules: <instance - at 0x7f410b057c20>
               args: <tuple - at 0x7f411be1e050>
               here: <ImplicitAcquisitionWrapper document-55286 at 0x7f40da8a2050>
               user: <ImplicitAcquisitionWrapper - at 0x7f40e101dc30>
               file_size: 529
               nothing: <NoneType - at 0x7ad050>
               container: <ImplicitAcquisitionWrapper document-55286 at 0x7f40da8a2050>
               request: <instance - at 0x7f40d8678518>
               wrapped_repeat: <SafeMapping - at 0x7f40d8b421b0>
               traverse_subpath: <list - at 0x7f40d9484c68>
               default: <object - at 0x7f411bdf4c20>
               loop: {...} (0)
               context: <ImplicitAcquisitionWrapper document-55286 at 0x7f40da8a2050>
               view: <SimpleViewClass from /home/zope/eggs/opengever.core-2017.6.8-py2.7.egg/opengever/bumblebee/browser/templates/bumblebeeoverlay.pt bumblebee-overlay-document at 0x7f40dbed3f10>
               translate: <function translate at 0x7f40e8ee4500>
               root: <ImplicitAcquisitionWrapper Zope at 0x7f40dab973c0>
               options: {...} (0)
               target_language: de
```

### PR DESCRIPTION
**Pull request by Rotonen**
* Added an empty document into the fixtures
* Ported the bumblebee overlay tests to integration tests
* Fixed the bumblebee document overlay date getter to allow for `None`

Closes #4723

### CODE REVIEW

File: opengever/testing/fixtures.py [hunk 1]
```diff
@@ -1042,6 +1042,12 @@ def create_treaty_dossiers(self):
                 )
             ))
 
+        self.register('empty_document', create(
+            Builder('document')
+            .within(self.dossier)
+            .titled(u'L\xe4\xe4r')
```
Comment (phgross) [inline opengever/testing/fixtures.py]: Theres a flag `preserved_as_paper` which has to be true, wenn creating an empty document through the form. I would set this value here to.

File: opengever/bumblebee/tests/test_overlay_adapter_document.py [hunk 2]
```diff
@@ -1,571 +1,425 @@
...
-
-        adapter = getMultiAdapter((document, self.request), IBumblebeeOverlay)
+        self.login(self.regular_user)
+        adapter = getMultiAdapter((self.document, self.request), IBumblebeeOverlay)
 
         self.assertEqual(
-            'http://nohost/plone/dossier-1/document-1/'
-            'bumblebee-open-pdf?filename=Testdokumaent.pdf',
-            adapter.get_open_as_pdf_url())
+            'http://nohost/plone/ordnungssystem/fuhrung/vertrage-und-vereinbarungen/'
```
Comment (phgross) [inline opengever/bumblebee/tests/test_overlay_adapter_document.py]: Using absolute_url() would be a bit safer for feature fixture changes 

```
self.assertEquals(
    '{}/bumblebee-open-pdf?filename=Vertraegsentwurf.pdf'.format(self.document.absolute_url(),
    adapter.get_open_as_pdf_url())
```

File: opengever/bumblebee/tests/test_overlay_adapter_document.py [hunk 3]
```diff
@@ -1,571 +1,425 @@
...
-                              u'example.docx'))
-
-        adapter = getMultiAdapter((document, self.request), IBumblebeeOverlay)
-
+        self.login(self.regular_user)
+        adapter = getMultiAdapter((self.document, self.request), IBumblebeeOverlay)
         self.assertIn(
-            'http://nohost/plone/dossier-1/document-1/editing_document',
-            adapter.get_checkout_url())
+            'http://nohost/plone/ordnungssystem/fuhrung/'
```
Comment (phgross) [inline opengever/bumblebee/tests/test_overlay_adapter_document.py]: see comment above.

File: opengever/bumblebee/tests/test_overlay_adapter_document.py [hunk 4]
```diff
@@ -1,571 +1,425 @@
...
-        dossier = create(Builder('dossier'))
-        document = create(Builder('document').within(dossier))
-
-        adapter = getMultiAdapter((document, self.request), IBumblebeeOverlay)
+        self.login(self.regular_user)
+        adapter = getMultiAdapter((self.document, self.request), IBumblebeeOverlay)
         self.assertEqual(
-            'http://nohost/plone/dossier-1/document-1/edit',
-            adapter.get_edit_metadata_url())
+            'http://nohost/plone/ordnungssystem/fuhrung/vertrage-und-vereinbarungen/dossier-1/document-12/edit',
```
Comment (phgross) [inline opengever/bumblebee/tests/test_overlay_adapter_document.py]: see comment above.

File: opengever/bumblebee/tests/test_overlay_adapter_document.py [hunk 5]
```diff
@@ -1,571 +1,425 @@
...
-
-        adapter = getMultiAdapter((document, self.request), IBumblebeeOverlay)
+        self.login(self.regular_user)
+        self.checkout_document(self.document)
+        adapter = getMultiAdapter((self.document, self.request), IBumblebeeOverlay)
         self.assertIn(
-            'http://nohost/plone/dossier-1/document-1/'
-            '@@checkin_without_comment',
-            adapter.get_checkin_without_comment_url())
+            'http://nohost/plone/ordnungssystem/fuhrung/'
```
Comment (phgross) [inline opengever/bumblebee/tests/test_overlay_adapter_document.py]: see comment above.

File: opengever/ech0147/tests/test_model.py [hunk 6]
```diff
@@ -110,9 +110,11 @@ def test_dossier_path_consists_of_base_path_and_dossier_id(self):
     def test_dossier_contains_documents_and_mails(self):
         documentish_types = ['opengever.document.document', 'ftw.mail.mail']
         dossier = Dossier(self.dossier, u'files')
+        # XXX - do not include documents without files - document vs. mail
```
[the remark this answers is not in the document]
Comment (Rotonen) [reply opengever/ech0147/tests/test_model.py]: The ech0147 exports of dossiers exclude documents without files, but the query to fetch the expected documents also contains mails, but mails do not have a file property to assert on.

Now a document without a file was added to the fixtures, but the query is a bit on the unintuitive side. The remedy is to refactor the split between documents and mails.

**Comment by Rotonen** [comment]
Ready for rereview:

* Empty document now marked as preserved as paper
* Prettied up the ugly query in the ech0147 dossier export tests
* Adapter tests now independent of fixture changes

**Pull request merged**